pkg/operator/ceohelpers/machine_api.go

```diff
@@ -0,0 +1,56 @@
+package ceohelpers
+
+import (
+	machinelistersv1beta1 "github.com/openshift/client-go/machine/listers/machine/v1beta1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/utils/pointer"
+)
+
+// MachineAPI a simple struct that helps determine if the Machine API is functional
+type MachineAPI struct {
+	hasMasterMachineInformerSyncedFn func() bool
+	masterMachineLister              machinelistersv1beta1.MachineLister
+	masterMachineSelector            labels.Selector
+}
+
+func NewMachineAPI(masterMachineInformer cache.SharedIndexInformer, masterMachineLister machinelistersv1beta1.MachineLister, masterMachineSelector labels.Selector) *MachineAPI {
```

Comment:
This is exactly what I wanted to avoid. I created a simple function but David didn't like it and wanted a struct. Now Joel wants an interface.

I made David happy by creating a struct. I think it would be very unfair if I wouldn't do the same for you :) Let me update the PR :)

Reply:
You should always feel free to reject feedback if you have good reason, this is by no means a strong opinion of mine on this one